addons/binding/org.openhab.binding.hs110/src/main/java/org/openhab/binding/hs110/internal/HS110HandlerFactory.java

```diff
@@ -0,0 +1,45 @@
+/**
+ * Copyright (c) 2014-2016 by the respective copyright holders.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.hs110.internal;
+
+import static org.openhab.binding.hs110.HS110BindingConstants.*;
+
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingTypeUID;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandlerFactory;
+import org.eclipse.smarthome.core.thing.binding.ThingHandler;
+import org.openhab.binding.hs110.handler.HS110Handler;
+
+/**
+ * The {@link HS110HandlerFactory} is responsible for creating things and thing
+ * handlers.
+ *
+ * @author Christian Fischer - Initial contribution
+ */
+public class HS110HandlerFactory extends BaseThingHandlerFactory {
+
+    @Override
+    public boolean supportsThingType(ThingTypeUID thingTypeUID) {
+        return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
+    }
+
+    @Override
+    protected ThingHandler createHandler(Thing thing) {
+
+        ThingTypeUID thingTypeUID = thing.getThingTypeUID();
+
+        if (thingTypeUID.equals(THING_TYPE_HS100)) {
+            return new HS110Handler(thing);
+        }
+        if (thingTypeUID.equals(THING_TYPE_HS110)) {
```

Comment:
Yes, currently it is very hard how to distinguish the HS110 and HS100 - i do only have the newer one available. The architecture to be built isn't clear at the moment. So i would leave this as a stub for somebody to step in. 

It _should_ work with the HS100 out of the box though.